appservice/src/intTest/groovy/nu/yona/server/DeviceTest.groovy

```diff
@@ -251,27 +253,29 @@ class DeviceTest extends AbstractAppServiceIntegrationTest
 		def richard = addRichard()
 
 		when:
-		def response = richard.devices[0].postOpenAppEvent(appService, richard.devices[0].operatingSystem, "0.0.1")
+		def appVersion = "9.9.9"
```

Comment:
I guess this app version is not interpreted at all. This could be reflected in the name of the test: "Try to post app opened event for a too old app version code". A comment could be added here that this app version is ignored.

Reply:
The version code is the deciding factor, but we don't communicate that to the user. That's why this is put in a variable and then used once in the request and then to assert the error message.